0000-update-glimmer-template-truthiness-semantics.md

```diff
@@ -0,0 +1,106 @@
+
+### {{or}}
+Takes at least two positional arguments. Raises an error if invoked with less than two arguments. It evaluates arguments left to right, returning the first one that is truthy (by javascript's definition of truthiness) or the right-most argument if all evaluate to falsy. This is equivalent to the {{or}} helper from ember-truth-helpers. Doing this would allow `{{or}}` to compile down to `||` and give us the correct type safety and narrowing that users would expect.
+
+### {{not}}
+Unary operator. Raises an error if invoked with more than one positional argument. If the given value evaluates to a truthy value (by javscript's definition of truthiness), the false is returned. If the given value evaluates to a falsy value (by javascript's definition of truthiness) then it returns true. This is equivalent to the {{not}} helper from ember-truth-helpers. Doing this would allow `{{not}}` to compile down to `!` and give us the correct type safety and narrowing that users would expect.
+
+### {{if}}
+Same implementation as current `if`. The only difference is when doing the check for truthiness it would follow javascript's truthiness semantics.
+
+### {{unless}}
+Same implementation as current `unless`. The only difference is when doing the check for truthiness it would follow javascript's truthiness semantics.
+
+### {{eq}}
+Binary operation. Throws an error if not called with exactly two arguments. Uses triple equal version of javascript equality. Doing this allows for `{{eq}}` to compile down to `===` and give us the correct type safety and narrowing that users would expect
+
+### {{neq}}
+Binary operation. Throws an error if not called with exactly two arguments. Uses triple not equal version of javascript equality. Doing this allows for `{{neq}}` to compile down to `!==` and give us the correct type safety and narrowing that users would expect
+
+### {{lt}}
+Binary operation. Throws an error if not called with exactly two arguments. Equivalent of `<` in Javascript.
+
+### {{lte}}
+Binary operation. Throws an error if not called with exactly two arguments. Equivalent of `<=` in Javascript.
+
+### {{gt}}
+Binary operation. Throws an error if not called with exactly two arguments. Equivalent of `>` in Javascript.
+
+### {{gte}}
+Binary operation. Throws an error if not called with exactly two arguments. Equivalent of `>=` in Javascript.
+
+### Migration Plan
+1. Ship comparison helpers `and`, `or`, `not` with existing equality and truthiness semantics (they currently match `if` and `unless` truthiness semantics)
+2. Ship types which match that and document why they work they way they do. Acknowledge that this wont work for TS users in they way they would expect and why.
+3. Ship ability to opt into new behavior for `and`, `or`, `not`, `if` and `unless` (Following JS truthiness semantics)
```

Comment:
How would that opt-in look like? Would be great if you could do more in detail for the migration story in general.